tobler/tests/test_interpolators.py

```diff
@@ -8,26 +8,35 @@
 from tobler.dasymetric import glm, glm_pixel_adjusted, masked_area_interpolate
 from tobler.area_weighted import area_interpolate
 
-p = quilt3.Package.browse("rasters/nlcd", "s3://quilt-cgs")
-p["nlcd_2011.tif"].fetch()
 
 local_raster = os.getcwd() + "/nlcd_2011.tif"
 
-sac1 = load_example("Sacramento1")
-sac2 = load_example("Sacramento2")
 
-sac1 = geopandas.read_file(sac1.get_path("sacramentot2.shp"))
-sac2 = geopandas.read_file(sac2.get_path("SacramentoMSA2.shp"))
+def datasets():
+
+    if not os.path.exists(local_raster):
+        p = quilt3.Package.browse("rasters/nlcd", "s3://quilt-cgs")
+        p["nlcd_2011.tif"].fetch()
+
+    sac1 = load_example("Sacramento1")
+    sac2 = load_example("Sacramento2")
+
+    sac1 = geopandas.read_file(sac1.get_path("sacramentot2.shp"))
+    sac2 = geopandas.read_file(sac2.get_path("SacramentoMSA2.shp"))
+
+    return sac1, sac2
```

Comment:
Should this return the raster so tests have it?

Reply:
the functions just use the path, so all it needs to do is check whether a raster is available at that location and it will be available when necessary

Reply:
where is [local_raster](https://github.com/pysal/tobler/blob/9ae93378be60159d1fa97970c0b9d4307309c80c/tobler/tests/test_interpolators.py#L44) defined?

Reply:
line 12

Reply:
nevermind - i see it :->